extensions/guacamole-auth-quickconnect/src/main/java/org/apache/guacamole/auth/quickconnect/QuickConnectDirectory.java

```diff
@@ -118,5 +136,54 @@ public String create(GuacamoleConfiguration config) throws GuacamoleException {
 
         return newConnectionId;
     }
+    
+    /**
+     * Checks the provided GuacamoleConfiguration to make sure the parameters
+     * in the configuration are allowed by the configuration.  If the
+     * allowed parameter configuration option is set, all parameters not
+     * specified will be removed from the config.  If the denied parameter
+     * configuration option is set, any parameters specified by that will be
+     * removed from the configuration.  If neither are set then all parameters
+     * are allowed.
+     * 
+     * @param config
+     *     The GuacamoleConfiguration containing parameters to check against
+     *     configured allowed and denied values.
+     * 
+     * @throws GuacamoleException 
+     *     If guacamole.properties cannot be parsed.
+     */
+    private void checkConfigParameters(GuacamoleConfiguration config) throws GuacamoleException {
```

Comment:
Should this instead be part of `QCParser` (and tested via `QCParserTest`)? Comments within `QCParser` look like this may have originally been intended, and it feels like ensuring `QCParser` always abides by the set of allowed/denied parameters would be a safer approach

Reply:
Yeah, probably so - I think I was being lazy and trying to avoid having to pull the `ConfigurationService` class/injection into the `QCParser` class, but it's probably worth it.

Reply:
Okay, moved it, with a couple of extra parameters to avoid having to pull `ConfigurationService` into the `QCParser` class.

Reply:
Also added unit tests...